master/internal/rm/kubernetesrm/request_workers.go

```diff
@@ -89,31 +134,72 @@ func (r *requestProcessingWorker) receiveDeleteKubernetesResources(
 
 	// If resource creation failed, we will still try to delete those resources which
 	// will also result in a failure.
-	if len(msg.podName) > 0 {
+	if podName := msg.podName; podName != nil {
```

Comment:
nit: maybe skip to lower the future merge conflict surface

Reply:
I think this change is valuable and I think we are already going to conflict pretty hard with pods to jobs, so keeping this the same